node/tests/sync_test.go

```diff
@@ -112,3 +112,104 @@ func TestSyncStartStopLightWithBridge(t *testing.T) {
 
 	assert.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, 40))
 }
+
+/*
+Test-Case: Sync a Full Node with a Bridge Node
+Steps:
+1. Create a Bridge Node(BN)
+2. Start a BN
+3. Check BN is synced to height 20
+4. Create a Full Node(FN) with a trusted peer
+5. Start a FN with a defined connection to the BN
+6. Check FN is synced to height 30
+*/
+func TestSyncFullWithBridge(t *testing.T) {
+	sw := swamp.NewSwamp(t, swamp.DefaultComponents())
+
+	bridge := sw.NewBridgeNode()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	t.Cleanup(cancel)
+
+	sw.WaitTillHeight(ctx, 20)
+
+	err := bridge.Start(ctx)
+	require.NoError(t, err)
+
+	h, err := bridge.HeaderServ.GetByHeight(ctx, 20)
+	require.NoError(t, err)
+
+	assert.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, 20))
+
+	addrs, err := peer.AddrInfoToP2pAddrs(host.InfoFromHost(bridge.Host))
+	require.NoError(t, err)
+
+	full := sw.NewFullNode(node.WithTrustedPeer(addrs[0].String()))
+	require.NoError(t, full.Start(ctx))
+
+	h, err = full.HeaderServ.GetByHeight(ctx, 30)
+	require.NoError(t, err)
+
+	assert.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, 30))
+}
+
+/*
+Test-Case: Sync a Light Node from a Full Node
+Pre-Requisites:
+- CoreClient is started by swamp
+- CoreClient has generated 20 blocks
+Steps:
+1. Create a Bridge Node(BN)
+2. Start a BN
+3. Check BN is synced to height 20
+4. Create a Full Node(FN) with a trusted peer
+5. Start a FN with a defined connection to the BN
+6. Check FN is synced to height 30
+7. Start a LN with a defined connection to the FN
+8. Start LN
+9. Check LN is synced to height 50
+*/
+func TestSyncLightWithFull(t *testing.T) {
+	sw := swamp.NewSwamp(t, swamp.DefaultComponents())
+
+	bridge := sw.NewBridgeNode()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	t.Cleanup(cancel)
+
+	sw.WaitTillHeight(ctx, 20)
+
+	err := bridge.Start(ctx)
+	require.NoError(t, err)
+
+	h, err := bridge.HeaderServ.GetByHeight(ctx, 20)
+	require.NoError(t, err)
+
+	assert.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, 20))
+
+	addrs, err := peer.AddrInfoToP2pAddrs(host.InfoFromHost(bridge.Host))
+	require.NoError(t, err)
+
+	full := sw.NewFullNode(node.WithTrustedPeer(addrs[0].String()))
+	require.NoError(t, full.Start(ctx))
+
+	h, err = full.HeaderServ.GetByHeight(ctx, 30)
+	require.NoError(t, err)
+
+	assert.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, 30))
+
+	addrs, err = peer.AddrInfoToP2pAddrs(host.InfoFromHost(full.Host))
+	require.NoError(t, err)
+	light := sw.NewLightNode(node.WithTrustedPeer(addrs[0].String()))
+
+	err = sw.Network.UnlinkPeers(bridge.Host.ID(), light.Host.ID())
```

Comment:
This test-case is covering the case, where a Light Node has no linkage to Bridge Node and syncs from a Full Node only